### PR TITLE
Add nixpkgs-fmt to shell and CI

### LINKED ISSUE #2980: Add nixpkgs-fmt to CI so that nix files are automatically formatted
Program itself is available in Nix under `pkgs.nixpkgs-fmt` and also nicely integrates with Emacs.
Reference: haskell formatting done here `./scripts/buildkite/check-stylish.sh`, something similar would most likely be needed

### PR DESCRIPTION
**Pull request by EncodePanda**
Fixes #2980 

One less thing to worry about.

Btw it integrates nicely with Emacs.

```
(use-package nixpkgs-fmt)
;; format all nix files on save
(add-hook 'nix-mode-hook 'nixpkgs-fmt-on-save-mode)
```

### CODE REVIEW

**Comment by nfrisby** [comment]
LGTM but I'll let @coot Approve -- he touches Nix more than I do 👍

**Review by nc6 — APPROVED** [review]
LGTM. Maybe you want to add a note to the style guide somewhere talking about the nix formatter

**Comment by coot** [comment]
I think @jbgi should review this PR. @jbgi is there a recommended formatter for `nix`?

File: scripts/buildkite/check-nixpkgs-fmt.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+nixpkgs-fmt `git ls-files -- '*.nix'`
```
Comment (jbgi) [inline scripts/buildkite/check-nixpkgs-fmt.sh:6]:
```suggestion
nixpkgs-fmt `git ls-files -- '*.nix' ':!:nix/sources.nix'`
```
```nix/sources.nix``` is generated by `niv` which will print a warning if the files has changed (thinking it is not up-to-date).

**Review by jbgi — CHANGES_REQUESTED** [review]
Small changed needed.
There is an alternative approach used in plutus that may be interesting, which integrate with hydra and provide a pre-commit hook via https://github.com/cachix/pre-commit-hooks.nix, cf.:
https://github.com/input-output-hk/plutus/blob/master/nix/tests/nixpkgs-fmt.nix
https://github.com/input-output-hk/plutus/blob/86ee2701586c26d0c85961533f5c82f1ee3ff076/shell.nix#L32-L38
cc/ @disassembler

**Review by jbgi — APPROVED** [review]

**Comment by EncodePanda** [comment]
bors merge

**Pull request merged**